sklearn/utils/_print_versions.py

```diff
@@ -0,0 +1,156 @@
+"""
+Utility methods to print system info for debugging
+
+adapted from :func:`pandas.show_versions`
+"""
+# License: BSD 3 clause
+
+import os
+import platform
+import sys
+import struct
+import subprocess
+import codecs
+import locale
+import importlib
+
+from .._build_utils import get_blas_info
+
+
+def get_sys_info():
+    """System information
+
+    Return
+    ------
+    sys_info : dict
+        system and Python version information
+
+    """
+
+    blob = []
+
+    # get full commit hash
+    commit = None
+    if os.path.isdir(".git") and os.path.isdir("sklearn"):
+        try:
+            pipe = subprocess.Popen('git log --format="%H" -n 1'.split(" "),
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+            so, serr = pipe.communicate()
+        except:
+            pass
+        else:
+            if pipe.returncode == 0:
+                commit = so
+                try:
+                    commit = so.decode('utf-8')
+                except ValueError:
+                    pass
+                commit = commit.strip().strip('"')
+
+    blob.append(('commit', commit))
+
+    try:
+        (sysname, nodename, release,
+         version, machine, processor) = platform.uname()
+        blob.extend([
+            ("python", '.'.join(map(str, sys.version_info))),
```

Comment:
I don't remember whether I commented already, but maybe `sys.executable` would be nice? This way we know whether it is a system/user/conda install. WDYT?